### ISSUE TITLE
Improve changelog generation workflow

### ISSUE THREAD

**Issue by J-Fields** [VSCodeVim/Vim#7460]
Currently we use some automated tool that doesn't seem to work right. Our changelog is generally terrible.

Probably, the best solution is to move to a semi-automated system where a PR is tagged with its changelog text - need to look into how to do this with GHA or a bot.

**Comment by macintacos** [comment]
I'm not gunna lie, I was about to open an issue about this 😅 

The most awesome thing (IMO obvi) to come in the latest release, implementing `inccommand` (#7416), was just listed without much fanfare in the "Merged pull requests" section. Would've been great to pull that up to the top - I'm sure a lot of folks will probably miss it.

What do y'all currently use to cut releases? I have some familiarity with a variety of release tooling so maybe I can help out.

**Comment by J-Fields** [comment]
@macintacos I agree entirely. I inherited a workflow that uses https://github.com/github-changelog-generator/github-changelog-generator. It's probably misconfigured as it does some wacky stuff, but I haven't taken the time to nail that down.

Any suggestions for workflows or tooling would be greatly appreciated; this stuff isn't my forte.

**Comment by macintacos** [comment]
I can't make you any promises, but I'm going to give this a shot. Please let me know if you'd rather take care of this!

**Comment by J-Fields** [comment]
For now, I'm just going to handwrite it in a simple format. Open to suggestions for ways to partially automate!
